src/access-controllers.js

```diff
@@ -0,0 +1,52 @@
+'use strict'
+
+const AccessControllerManifest = require('./access-controller-manifest')
+const IPFSAccessController = require('./ipfs-access-controller')
+const OrbitDBAccessController = require('./orbitdb-access-controller')
+
+let supportedTypes = {
+  'ipfs': IPFSAccessController,
+  'orbitdb': OrbitDBAccessController,
+}
+
+const getHandlerFor = (type) => {
+  if (!AccessControllers.isSupported(type)) {
+    throw new Error(`AccessController type '${type}' is not supported`)
+  }
+  return supportedTypes[type]
+}
+
+class AccessControllers {
+  static isSupported (type) {
+    return Object.keys(supportedTypes).includes(type)
+  }
+
+  static addAccessController (options) {
```

Comment:
Should this be `(type, options)` and require the user to explicitly give the type the given AC handles (which they define anyway in their custom controller)?

Reply:
We do this in idenitty-provider but I do prefer if the ac itself is self-describing rather than passing in type separately...do you think its clearer to explicitly say the type?

Reply:
I would agree with what you said, and doing so enforces to make sure the user have defined the type in their custom AC. Should be good if we add an existence check and throw an error if `static type () { ... }` is not defined in the custom AC or return non-string. What do you think?

Reply:
That works, we have an existence check but good to check for string

Reply:
:+1: